profiler/src/ProfilerEngine/Datadog.Profiler.Native.Linux/LinuxStackFramesCollector.cpp

```diff
@@ -347,14 +329,69 @@ std::int32_t LinuxStackFramesCollector::CollectCallStackCurrentThread()
     }
 }
 
-void LinuxStackFramesCollector::CollectStackSampleSignalHandler(int32_t signal)
+bool LinuxStackFramesCollector::CanCollect(int32_t threadId, pid_t processId) const
+{
+    // on OSX, processId can be equal to 0. https://sourcegraph.com/github.com/dotnet/runtime/-/blob/src/coreclr/pal/src/exception/signal.cpp?L818:5&subtree=true
+    // Since the profiler does not run on OSX, we leave it like this.
+    auto* currentThreadInfo = _pCurrentCollectionThreadInfo;
+    return currentThreadInfo != nullptr && currentThreadInfo->GetOsThreadId() == threadId && processId == _processId;
+}
+
+void LinuxStackFramesCollector::CallOrignalHandler(int32_t signal, siginfo_t* info, void* context)
+{
+    static thread_local bool alreadyExecuted = false;
```

Comment:
I think `executing` or `isExecuting`  would better reflect the intent (preventing reentrancy)

Reply:
✔️